exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchClient.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  /** @return true if all bulk records where flushed successfully */
+  public boolean flush() {
+    boolean success = true;
+    if (bulkRequest.numberOfActions() > 0) {
+      try {
+        final BulkResponse responses = client.bulk(bulkRequest, RequestOptions.DEFAULT);
+        success = checkBulkResponses(responses);
+      } catch (IOException e) {
+        throw new ElasticsearchExporterException("Failed to flush bulk", e);
+      }
+
+      if (success) {
+        // all records where flushed, create new bulk request, otherwise retry next time
+        bulkRequest = new BulkRequest();
+      }
+    }
+
+    return success;
+  }
+
+  private boolean checkBulkResponses(final BulkResponse responses) {
+    boolean failed = false;
+
+    for (BulkItemResponse response : responses) {
+      if (response.isFailed()) {
+        failed = true;
+        log.warn("Failed to flush bulk request {}", response.getFailureMessage());
+      }
+    }
+
+    return !failed;
+  }
+
+  public boolean shouldFlush() {
+    return bulkRequest.numberOfActions() >= configuration.bulk.size;
+  }
+
+  /** @return true if request was acknowledged */
+  public boolean putIndexTemplate(final ValueType valueType) {
+    final String templateName = indexPrefixForValueType(valueType);
+    final String filename = indexTemplateForValueType(valueType);
+    return putIndexTemplate(templateName, filename);
+  }
+
+  /** @return true if request was acknowledged */
+  public boolean putIndexTemplate(final String templateName, final String filename) {
+    final Map<String, Object> template;
+    try (InputStream inputStream = ElasticsearchExporter.class.getResourceAsStream(filename)) {
+      if (inputStream != null) {
+        template = XContentHelper.convertToMap(XContentType.JSON.xContent(), inputStream, true);
+      } else {
+        throw new ElasticsearchExporterException(
+            "Failed to find index template in classpath " + filename);
+      }
+    } catch (IOException e) {
+      throw new ElasticsearchExporterException(
+          "Failed to load index template from classpath " + filename, e);
+    }
+
+    // update prefix in template in case it was changed in configuration
+    template.put("index_patterns", Collections.singletonList(templateName + "-*"));
+
+    final PutIndexTemplateRequest request =
+        new PutIndexTemplateRequest(templateName).source(template);
+
+    return putIndexTemplate(request);
+  }
+
+  /** @return true if request was acknowledged */
+  private boolean putIndexTemplate(final PutIndexTemplateRequest putIndexTemplateRequest) {
+    try {
+      return client
+          .indices()
+          .putTemplate(putIndexTemplateRequest, RequestOptions.DEFAULT)
+          .isAcknowledged();
+    } catch (IOException e) {
+      throw new ElasticsearchExporterException("Failed to put index template", e);
+    }
+  }
+
+  // used in tests
+  protected Map<String, Object> get(Record<?> record) {
```

Comment:
If it's used strictly in tests, I would extract to a test client class which subclasses this, so (ideally) all code here is tested, and anything else is explicitly outside.

Reply:
I agree and will change it accordingly.